### PR TITLE
chore: upgrade rehype-katex with ESM support, update docs

### PR DESCRIPTION
**Pull request by pranabdas**
## Motivation

Since Docusaurus now supports importing ESM modules dynamically, we can upgrade the `retype-katex` to the latest version. I have updated the related documentation and `docusaurus.confing.js` accordingly.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests)?

Yes.

## Test Plan

Please see the deploy preview - <https://deploy-preview-6320--docusaurus-2.netlify.app/docs/markdown-features/math-equations/>

### CODE REVIEW

**Comment by Josh-Cena** [comment]
FWIW, I'm fine with updating it on our website, but not for docs—at least, not as the main content, but more as a supplementary. Those who need KaTeX are usually those not well-versed in JS and giving them such a complicated instruction is kinda overwhelming. The `@ts-expect-error` is also optional.

**Comment by pranabdas** [comment]
@Josh-Cena Shall I revert the doc and add a section describing optional ESM module import?

**Comment by Josh-Cena** [comment]
Yep, that sounds good! I think we mentioned somewhere that we don't suppose ESM; we can extend that and say "hey, you can use ESM, but not in the easiest way".

**Comment by slorber** [comment]
We should move everything inside `createConfig` if we need to import ESM modules, that's much more straightforward than modifying a static config later in a weird way.

And we don't necessarily need to pass plugins, wouldn't it work with just module names and let Docusauurs import those?

```diff
rehypePlugins: [katex],
rehypePlugins: ["katex"],
```

File: website/docusaurus.config.js [hunk 1]
```diff
@@ -261,7 +260,7 @@ const config = {
           showLastUpdateAuthor: true,
           showLastUpdateTime: true,
           remarkPlugins: [math, [npm2yarn, {sync: true}]],
-          rehypePlugins: [katex],
+          rehypePlugins: [],
```
Comment (pranabdas) [inline website/docusaurus.config.js:263]:
```
rehypePlugins: [],
```
I left it empty here in case there is another rehype plugin (which is not ESM).

**Pull request merged**